pkg/providers/azure.go

```diff
@@ -165,17 +167,34 @@ func (sc *SetUpCmd) CreateServicePrincipal() error {
 	return nil
 }
 
-func (sc *SetUpCmd) assignSpRole() error {
+func (sc *SetUpCmd) assignSpRole(ctx context.Context) error {
```

Comment:
Can we add a unit test case for this just to make sure it works via the SDK. ?

Reply:
Testing via the SDK involves interactions with Azure, which requires the proper setup of credentials, role assignments, and permissions, and this can be tricky to mock or simulate in a local testing environment.